### PR TITLE
Mod to user model chef menu

### LINKED ISSUE #21: BE: Modifications to the user model for chef + menu items
- Menu items model 
- a flag for is_chef on the user 

API routes:
- create a new menu items for a user (toggle the chef field)

### PR DESCRIPTION
**Pull request by giftofgrub**
Added a new script to create database.
Just run `pipenv run createdb`

I ran into an issue with the middleware @jwt_required. I created a workaround for now to jut get it running. What I did was get the Auth headers, extract the token from this, and then decode the token.

With the workaround, it should be able to generate new meal items when a POST is sent to "/meal_items"

### CODE REVIEW

**Comment by giftofgrub** [comment]
nvm the above comment. I was able to fix it

File: README.md [hunk 1]
```diff
@@ -16,12 +16,6 @@
...
-```
-python
-from app import create_app()
-app = create_app()
-app.app_context().push()
-from app import db
-db.create_all()
-exit()
+```sh
+pipenv run createdb
```
Comment (mssiegel) [inline README.md:17]: i think these lines should replace whats in step 4 under section "Adding Postgres database" in readme

File: server/api/user_handler.py [hunk 2]
```diff
@@ -1,8 +1,7 @@
-from models import db
 from models.User import User, UserSchema
 from flask import request, Response, json
 from flask_restful import Resource
-from flask_jwt_extended import create_access_token
+from flask_jwt_extended import create_access_token, get_jwt_identity
```
Comment (mssiegel) [inline server/api/user_handler.py]: get_jwt_identity is imported but not used

File: server/app.py [hunk 3]
```diff
@@ -1,14 +1,15 @@
-from flask import Flask
+from flask import Flask, request
```
Comment (mssiegel) [inline server/app.py]: request is imported but not used

File: server/models/MealItem.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    def __init__(self, userId, name, description=""):
+        self.userId = userId
+        self.name = name
+        self.description = description
+
+    def __repr__(self):
+        return f"<Meal #{self.id}: {self.name}>"
+
+    def save_to_database(self):
```
Comment (mssiegel) [inline server/models/MealItem.py]: MealItem and User models have the same save_to_database method. Is there a way to write this once and somehow have it used by both models.

File: server/api/login_handler.py [hunk 5]
```diff
@@ -24,9 +24,11 @@ def post(self):
         user = User.authenticate(email, password)
 
         if user:
+            token = create_refresh_token(user.to_dict())
```
Comment (mssiegel) [inline server/api/login_handler.py]: Token does not need to contain entire user object. it only needs to have user id.
`create_refresh_token(user.to_dict())` should be changed to `create_refresh_token({"id": user.id})`

File: server/api/meal_item_handler.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        try:
+            new_meal = MealItem(curr_user.id, name, description)
+            new_meal.save_to_database()
+            curr_user.chef_flag_true()
+            curr_user.save_to_database()
+            data = {
+                "message": "Created Meal Item",
+                "meal_item": new_meal.name
+            }
+            return Response(
```
Comment (mssiegel) [inline server/api/meal_item_handler.py]: return of Responses are all similar code and can be extracted to reusable function. 
[See my PR's ](https://github.com/hatchways/team-bluejay/pull/29/files)usage of function "custom_response"

File: server/models/MealItem.py (hunk 4 above)
Comment (giftofgrub) [reply server/models/MealItem.py]: You mean like helper function? Or, a separate class that they both inherit from?

File: server/models/User.py [hunk 7]
```diff
@@ -9,32 +9,22 @@ class User(db.Model):
...
     def __init__(self, name, email, password):
         if len(password) < 6:
             raise ValueError
         self.name = name
         self.email = email
         self.password = bcrypt.generate_password_hash(password).decode('UTF-8')
+        self.isChef = False
 
     def __repr__(self):
         return f"<User #{self.id}: {self.username}, {self.email}>"
```
Comment (mssiegel) [inline server/models/User.py]: in `__repr__`: `{self.username}` should be `{self.name}`

File: server/createdb.py [hunk 8]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env python
+
+from app import create_app
+app = create_app()
+app.app_context().push()
+from app import db
+db.create_all()
+print("just ran createdb.py")
+exit()
```
Comment (skassam21) [inline server/createdb.py:9]: This is a script right? Just maybe add it to a scripts folder, and use if __name__ == "__main__"

File: server/models/MealItem.py [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+from marshmallow import Schema
+
+
+class MealItem(db.Model):
+    __tablename__ = 'meal_items'
+
+    id = db.Column(db.Integer, primary_key=True)
+    userId = db.Column(
+        db.Integer,
+        db.ForeignKey('users.id', ondelete='CASCADE'),
```
Comment (skassam21) [inline server/models/MealItem.py:11]: Just want to double check - this ondelete="CASCADE" means when the user is deleted, we remove the meal items right?

**Review by skassam21 — COMMENTED** [review]
Two small things

File: server/models/MealItem.py (hunk 9 above)
Comment (giftofgrub) [reply server/models/MealItem.py:11]: Yes

**Review by mssiegel — CHANGES_REQUESTED** [review]
Really great work. 
I was able to create menu items.
We don't have a route for deleting users so I went into database directly to delete a user. Doing so automatically deleted all their menu items

Issues/Bugs:
-running `createdb.py script` didn't work on windows computer
-currently trying to create same user twice throws a weird error. it also causes subsequent successful new user ids to skip a number.

**Review by mssiegel — APPROVED** [review]

**Pull request merged**